src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -85,13 +85,13 @@ internal partial class Binder
                 }
             }
 
-            ReportDiagnosticsIfObsolete(diagnostics, conversion, syntax, hasBaseReceiver: false);
-
             if (conversion.IsMethodGroup)
             {
                 return CreateMethodGroupConversion(syntax, source, conversion, isCast: isCast, conversionGroupOpt, destination, diagnostics);
             }
 
+            ReportDiagnosticsIfObsolete(diagnostics, conversion, syntax, hasBaseReceiver: false);
```

Comment:
I moved this from line 88 to 93 to avoid repeated diagnostic.

Reply:
Moving to here is fine. We should leave it though, because there could be some other user-defined conversion operator involved that doesn't have anything to do with a method group. Please do add a comment to mention that obsolete diagnostics are reported as part of creating the method group conversion though.

---
In reply to: [491224318](https://github.com/dotnet/roslyn/pull/47332#discussion_r491224318) [](ancestors = 491224318)

Reply:
```suggestion
            // Obsolete diagnostics for method group are reported as part of creating the method group conversion.
            ReportDiagnosticsIfObsolete(diagnostics, conversion, syntax, hasBaseReceiver: false);
```